### DISCUSSION TITLE
[Question] Are parameters supported for ALTER TABLE statements?

### DISCUSSION THREAD

**Discussion by Endominus** [rusqlite/rusqlite#1176] [Questions] [answered]
I've been trying to execute the below command for a couple hours now, using named parameters and positional parameters, encapsulating the parameters in parentheses as some of the documentation seems to do and not. 

```sql
ALTER TABLE entries ADD COLUMN :name :type;
```
```rust
self.conn.execute("ALTER TABLE entries ADD COLUMN :name :type;", &[(":name", field.name.as_str()), (":type", type_full)]);
```

I get a syntax error at the :name (or opening parenthesis if using that). I've tried executing using the execute function and preparing a statement to execute separately. Given that similar code works with the `INSERT` command, and when hardcoding the name and type above, I can only guess that there's some issue in executing `ALTER TABLE` with parameters but there doesn't seem to be any note of this in the documentation. It's a statement, per the sqlite docs, and there doesn't seem to be a different specialized statement for table-altering commands in the rusqlite docs as far as I found.

Am I just blind and missing something obvious? Thanks in advance!

**Comment by thomcc** [comment, marked as answer]
Nope, this is not supported. One way of thinking of this is that the value parameters evaluate are always expressions (in this case one evaluating to a string), and not identifiers. Whereas the column name must be an identifier — perhaps you can think of it analgously to `let "foo" = 3`.

That said, this explanation is somewhat handwavey and is probably not *really* accurate — the true answer is probably closer to "it's a syntax error to put a parameter there", and I'm making a guess to why that might be. More generally parameterization tends not to be supported in the data definition sublanguage (e.g. the `CREATE`/`ALTER`/etc), although it may be supported in some cases I'm forgetting — definitely not for non-expression syntax elements like column/table names/types though.

**Reply by thomcc** [reply]
(P.S. sorry for converting this to a discussion — I think this is more of a SQL usage question than a rusqlite question, though)

**Reply by Endominus** [reply]
No problem, and thanks for letting me know. It's unfortunate but not unworkable - I'll just have to replace substrings manually.

**Reply by thomcc** [reply]
Yep, that's the best option. Sorry.
